src/server/qgsrequesthandler.h

```diff
@@ -96,33 +96,26 @@ class QgsRequestHandler
     //! @note not available in Python bindings
     virtual void setGetCoverageResponse( QByteArray* ba ) = 0;
 
-    virtual void setDefaultHeaders() {}
-
     //! Set an HTTP header
     virtual void setHeader( const QString &name, const QString &value ) = 0;
 
     //! Remove an HTTP header
-    virtual int removeHeader( const QString &name ) = 0;
-
-    //! Delete all HTTP headers
-    virtual void clearHeaders() = 0;
+    virtual void removeHeader( const QString &name ) = 0;
 
     //! Append the bytestream to response body
     virtual void appendBody( const QByteArray &body ) = 0;
 
-    //! Clears the response body
-    virtual void clearBody() = 0;
```

Comment:
So, if there is a good reason to not let the client code access the headers directly in write mode, it's mandatory to keep the getter/setters and add the missing keys() method. Why?
Because passing a read-only (or copy) object to python bindings is not what the developer expects: if I get a dictionary object in Python I expect I can alter it. Passing a tuple of tuples or some frozen struct doesn't seem the right thing to do either. 
Remember: I want to be able to create/edit/delete/retrieve individual header entries from client code and plugins.